Game.playground/Contents.swift

```diff
@@ -0,0 +1,335 @@
+
+// MARK: - Random names
+class RandomName {
+    static let names = ["Liam", "Olivia", "Noah", "Emma", "Oliver", "Ava", "William", "Sophia", "Elijah", "Isabella", "James", "Charlotte"]
+    
+    static func get() -> String {
+        return "\(names.randomElement()!) \(Int.random(in: 1...255))"
+    }
+}
+// MARK: - Character classes
+class Player {
+    var name: String
+    var health: Int
+    var damage: Int
+    
+    var team: Team
+    
+    init(name: String, health: Int, damage: Int, team: Team) {
+        self.health = health
+        self.name = name
+        self.damage = damage
+        self.team = team
+    }
+    
+    func die() {
+        print("\(name) (\(team.name)) is dead now.")
+        team.remove(player: self)
+    }
+    
+    func info() -> String {
+        return "\(name): hp = \(health), dmg = \(damage)"
+    }
+}
+
+class Warrior: Player {
+    var armor: Int
+    
+    init(name: String, health: Int, damage: Int, armor: Int, team: Team) {
+        self.armor = armor
+        super.init(name: name, health: health, damage: damage, team: team)
+    }
+    
+    func attack(player: Player) {
+        if player is Warrior {
+            let enemy = player as! Warrior
+            enemy.health -= (self.damage - enemy.armor)
+        }
+        if player is Mage {
+            let enemy = player as! Mage
+            enemy.health -= self.damage
+        }
```

Comment:
Чтобы не делать каста с восклицательным, можно вот так:
```suggestion
        if let warrior = player as? Warrior {
            warrior.health -= (self.damage - warrior.armor)
        } else if let mage = player as? Mage {
            mage.health -= self.damage
        }
```